libavfilter/vf_drawbox.c

```diff
@@ -273,9 +291,31 @@ static int filter_frame(AVFilterLink *inlink, AVFrame *frame)
 {
     DrawBoxContext *s = inlink->dst->priv;
     int xb = s->x, yb = s->y;
+    const AVDetectionBBoxHeader *header = NULL;
+    const AVDetectionBBox *bbox;
+    AVFrameSideData *sd;
+    int loop = 1;
+
+    sd = av_frame_get_side_data(frame, AV_FRAME_DATA_DETECTION_BBOXES);
+    if (s->box_source == AV_FRAME_DATA_DETECTION_BBOXES && sd) {
```

Comment:
how about the case: s->box_source == AV_FRAME_DATA_DETECTION_BBOXES && !sd

loop is 1.

Reply:
that means, we set box_source but no AVDetecionBBox in side_data, and it will draw a box around the pic, as drawbox always do with the unchanged code.
would this be OK?

Reply:
it doesn't make sense. I don't need a box when no object is detected.

Reply:
modified the code as we talked